### ISSUE TITLE
[Bug]: Atuin flags Starship "background processes" flag after each command.

### ISSUE THREAD

**Issue by jbiason** [atuinsh/atuin#1877]
### What did you expect to happen?

Starship have a flag to show if there are any processes running in the background (e.g., when you run a command with `&` or using Ctrl+Z + `bg`). I use this a lot, specially when editing code and leaving VIM running in the background.

I'd expect this to not change

### What happened?

Right now, after each command, Starship thinks there is a new process running in the background.

![image](https://github.com/atuinsh/atuin/assets/228850/4a6bec25-ae3e-4256-9f49-d295db57aebb)

### Atuin doctor output

```yaml
atuin:
  version: 18.1.0
  sync: null
shell:
  name: bash
  plugins:
  - atuin
  - bash-preexec
system:
  os: Ubuntu
  arch: x86_64
  version: '22.04'
  disks:
  - name: /dev/nvme0n1p2
    filesystem: ext4
  - name: /dev/nvme0n1p2
    filesystem: ext4
  - name: /dev/nvme0n1p1
    filesystem: vfat
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jbiason** [comment]
(I do understand that this happens 'cause Atuin is left updating the history just after the command ended, but I think there should be an option to make this "sync" and not "async".)

**Comment by ellie** [comment]
>  but I think there should be an option to make this "sync" and not "async"

I'm afraid this won't be implemented, as remote history sync can sometimes take a few seconds. The experience would be terrible.

This will be resolved by #1661, which I currently have a draft branch for
